pkg/scalers/prometheus_scaler.go

```diff
@@ -149,6 +149,10 @@ func (s *prometheusScaler) ExecutePromQuery() (float64, error) {
 	}
 	r.Body.Close()
 
+	if !(r.StatusCode >= 200 && r.StatusCode <= 299) {
```

Comment:
Just curious some 3xx (Redirects etc) are not relevant in here?

Reply:
As per [api docs](https://prometheus.io/docs/prometheus/latest/querying/api/#format-overview), it returns 2xx.

> The API response format is JSON. Every successful API request returns a 2xx status code.

Reply:
Thanks for the confirmation!